docs/tutorials/two_screens.rst

```diff
@@ -0,0 +1,493 @@
+
+.. jupyter-execute::
+
+    normal2 = CylindricalRepresentation(1., 90.*u.deg - xi2, 0.).to_cartesian()
+
+    screen2 = Screen1D(normal=normal2,
+                       p=pos2,
+                       v=v_lens2,
+                       magnification=magnification2)
+
+
+Generating observations
+=======================
+
+Use the :py:meth:`~screens.screen.Screen.observe` method to generate two sets
+of optical paths: one for radiation scattered only by the main screen
+(resulting in the main parabola) and one for radiation scattered by both
+screens (yielding the millisecond feature).
+
+.. jupyter-execute::
+
+    obs1 = telescope.observe(
+        source=screen1.observe(source=pulsar, distance=d_p-d_s1),
+        distance=d_s1)
+
+    obs2 = telescope.observe(
+        source=screen1.observe(
+            source=screen2.observe(source=pulsar, distance=d_p-d_s2),
+            distance=d_s2-d_s1),
+        distance=d_s1)
+
+The :py:class:`~screens.screen.Screen1D` class assumes that the linear features
+that cause the images on the lens continue indefinitely. The lens that causes
+the millisecond feature, however, is found to be limited in extent, not
+producing any scatterings beyond a certain point on the sky. To model this, we
+have to use a little hack: we can select optical paths in ``obs2`` based on
+their positions at the main screen. For these positions to be available, they
+first need to be computed. This is triggered by the first line, which computes
+the geometric delays of the optical paths, for which the positions of the
+scattering points on both screens need to be calculated.
+
+.. jupyter-execute::
+
+    obs2.tau
+    bool_on_lens2 = obs2.source.pos.x.squeeze() < 7. * u.au
+
+Using the ``tau``, ``taudot``, and ``brightness`` attributes of ``obs1`` and
+``obs2``, we can find the geometric delays of the optical paths (at the
+reference time), their time derivatives, and their complex magnifications. We
+combine the two sets of optical paths into a single list, using only the ones
+from ``obs2`` that were selected above using ``bool_on_lens2``.
+
+.. jupyter-execute::
+
+    tau0 = np.hstack([obs1.tau.ravel(),
+                      obs2.tau.ravel()[bool_on_lens2]])
+    taudot = np.hstack([obs1.taudot.ravel(),
+                        obs2.taudot.ravel()[bool_on_lens2]])
+    brightness = np.hstack([obs1.brightness.ravel(),
+                            obs2.brightness.ravel()[bool_on_lens2]])
+
+Plot the optical paths in conjugate variable space (the Doppler shift can be
+expressed as :math:`f_\mathrm{D} = f_\mathrm{obs} \dot{\tau}`,
+where :math:`f_\mathrm{obs}` is the observing frequency).
+This figure should correspond to the norm of the conjugate wavefield,
+showing the arclet apices with their associated magnifications.
+
+.. jupyter-execute::
+
+    fd_all = f_obs * taudot
+
+    plt.figure(figsize=(12., 8.))
+
+    plt.scatter(fd_all.to(u.mHz), tau0.to(u.us),
```

Comment:
It might be an idea to enable `astropy.visualization.quantity_support()` on top - right now, the units on X, Y are simply ignored, which is fine given that you label things correctly, but one might as well set it up such that when the user plots something else on top it will work.

(Of course, you could then also *not* to `.to(u.mHz)` etc and then insert `fd_all.unit` in the label below, as you do already for the dynamic spectrum.)

Reply:
I left this is it was